src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java

```diff
@@ -459,11 +468,12 @@ public void updateState(boolean reactiveLimits, boolean writeSlackBus) {
         updateGeneratorsState(voltageControl ? calculatedQ + loadTargetQ : generationTargetQ, reactiveLimits);
 
         // update load power
-        double factor = initialLoadTargetP != 0 ? loadTargetP / initialLoadTargetP : 1;
+        double factorP = initialLoadTargetP != 0 ? loadTargetP / initialLoadTargetP : 1;
+        double factorQ = initialLoadTargetQ != 0 ? loadTargetQ / initialLoadTargetQ : 1;
         for (Load load : loads) {
             load.getTerminal()
-                    .setP(load.getP0() >= 0 ? factor * load.getP0() : load.getP0())
-                    .setQ(load.getQ0());
+                    .setP(load.getP0() >= 0 ? factorP * load.getP0() : load.getP0())
+                    .setQ(load.getQ0() >= 0 ? factorQ * load.getQ0() : load.getQ0());
```

Comment:
@annetill: why do we distinguish positive and negative loads?

Reply:
I think these tests are not useful, I have to check.